lib/tasks/data_migration.rake

```diff
@@ -93,4 +93,66 @@ namespace :data_migration do
 
     puts "Total number of appointments marked as 'manual' = #{number_of_appointments_marked_manual}"
   end
+
+  desc "Create automatic appointment for defaulters so that they show up in the Overdue List"
+  task create_automatic_appointment_for_defaulters: :environment do
+    last_bp_older_than_one_month = ->(p) { p.latest_blood_pressure.blank? || p.latest_blood_pressure.device_created_at < 1.month.ago }
+
+    patient_is_hypertensive = ->(p) { p.latest_blood_pressure.present? && p.latest_blood_pressure.hypertensive? }
+
+    patient_has_cardio_vascular_history = ->(p) { p.medical_history.present? && p.medical_history.indicates_risk? }
+
+    defaulters = Patient.select do |p|
+      p.appointments.count == 0 &&
+        last_bp_older_than_one_month.call(p) &&
+        (patient_is_hypertensive.call(p) || patient_has_cardio_vascular_history.call(p) || p.prescription_drugs.present?)
+    end
+
+    puts "Found #{defaulters.count} defaulters"
+
+    if defaulters.blank? || defaulters.empty?
+      puts "No defaulters found. Aborting task."
+      return
+    end
+
+    puts "Processing #{defaulters.count} defaulters..."
+
+    processed_defaulters_count = 0
+    unprocessed_or_errored_defaulters_count = 0
+
+    defaulters.each do |defaulter|
+      if defaulter.latest_blood_pressure.blank?
+        puts "Failed to find latest Blood Pressure reading for defaulter #{defaulter.id}. Skipping patient..."
+        unprocessed_or_errored_defaulters_count += 1
+        next
+      end
+
+      latest_bp_facility_id = defaulter.latest_blood_pressure.facility_id
+      app_creation_time = Time.now
```

Comment:
Use latest blood pressures `device_created_at` and `app_creation_time + 30.days` for `app_creation_time` and `app_scheduled_date`